smslibrary/src/main/java/com/eis/smslibrary/SMSMessage.java

```diff
@@ -70,6 +105,32 @@ public SMSPeer getPeer() {
         return peer;
     }
 
+    /**
+     * Indicates whether some other object is "equal to" this one.
+     *
+     * @param o the reference object with which to compare.
+     * @return {@code true} if this object is the same as the o argument; {@code false} otherwise.
+     */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```

Comment:
if (!(o instanceof SMSMessage)) return false;
is the same but more elegant and readable

Reply:
The [specification](https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#equals-java.lang.Object-) of `Object.equals()` says
> It is _symmetric_: for any non-null reference values `x` and `y`, `x.equals(y)` should return `true` if and only if `y.equals(x)` returns `true`.

`o instanceof subclass` returns `false` when `o` is an instance of a superclass;
`o instanceof superclass` returns `true` when `o` is an instance of a subclass.

Reply:
because whe you use a subclass the cast is implicit
i don't know the implicatins of these two methods
but until I found an overwhelming evidence in my favor you are right

Reply:
Ok, can I mark this conversation as resolved?

Reply:
ok